oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobContainerProvider.java

```diff
@@ -171,176 +152,99 @@ public String getContainerName() {
-                               SharedAccessBlobHeaders optionalHeaders) throws InvalidKeyException, StorageException {
-        return optionalHeaders == null ? blob.generateSharedAccessSignature(policy, null) :
-                blob.generateSharedAccessSignature(policy,
-                        optionalHeaders, null, null, null, true);
+    private BlobContainerClient getBlobContainerFromConnectionString() {
+        return new BlobContainerClientBuilder()
+                .connectionString(getAzureConnectionString())
+                .containerName(getContainerName())
+                .buildClient();
     }
 
     private boolean authenticateViaServicePrincipal() {
         return StringUtils.isBlank(azureConnectionString) &&
                 StringUtils.isNoneBlank(accountName, tenantId, clientId, clientSecret);
     }
 
-    private class TokenRefresher implements Runnable {
-        @Override
-        public void run() {
-            try {
-                log.debug("Checking for azure access token expiry at: {}", LocalDateTime.now());
-                OffsetDateTime tokenExpiryThreshold = OffsetDateTime.now().plusMinutes(5);
-                if (accessToken.getExpiresAt() != null && accessToken.getExpiresAt().isBefore(tokenExpiryThreshold)) {
-                    log.info("Access token is about to expire (5 minutes or less) at: {}. New access token will be generated",
-                            accessToken.getExpiresAt().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
-                    AccessToken newToken = clientSecretCredential.getTokenSync(new TokenRequestContext().addScopes(AZURE_DEFAULT_SCOPE));
-                    log.info("New azure access token generated at: {}", LocalDateTime.now());
-                    if (newToken == null || StringUtils.isBlank(newToken.getToken())) {
-                        log.error("New access token is null or empty");
-                        return;
-                    }
-                    // update access token with newly generated token
-                    accessToken = newToken;
-                    storageCredentialsToken.updateToken(accessToken.getToken());
-                }
-            } catch (Exception e) {
-                log.error("Error while acquiring new access token: ", e);
-            }
-        }
+    private ClientSecretCredential getClientSecretCredential() {
+        return new ClientSecretCredentialBuilder()
+                .clientId(clientId)
+                .clientSecret(clientSecret)
+                .tenantId(tenantId)
+                .build();
     }
 
-    @Override
-    public void close() {
-        new ExecutorCloser(executorService).close();
-        log.info("Refresh token executor service shutdown completed");
+    @NotNull
+    private BlobContainerClient getBlobContainerFromServicePrincipals(String accountName, String clientId, String clientSecret, String tenantId, RequestRetryOptions retryOptions) throws DataStoreException {
```

Comment:
here you are parsing the `accountName` however at line [218](https://github.com/apache/jackrabbit-oak/pull/1861/files#diff-5009e27269a530925293e139d27368a1bc75abb12217bbff287386c4586073b2R218) for method `getBlobContainerFromConnectionString()` you are using the `getContainerName()`.
please use the same logic for both methods